File: March-2023/C++/NumberOfZeroFilledSubarrays.cxx
// 2348. Number of Zero-Filled Subarrays
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long result = 0;
        
        int n = nums.size();
        
        int countSubArray = 0;
        
        for(int i = 0; i<n; i++) {
            
            if(nums[i] == 0)
                countSubArray += 1;
            else
                countSubArray = 0;
            
            result += countSubArray;
            
        }
        
        return result;
        
    }
};

File: March-2023/Java/NumberOfZeroFilledSubarrays.java
// Number of Zero-Filled Subarrays
// Algorithm
// Get the count of continuous zero then add it in resulta
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long result=0;
        int count=0;
        for(int i=0; i<nums.length; i++){
            if(nums[i]==0){
                count++;
            }
            else{
                count=0;
            }
            result+=count;
        }
        return result;
    }
}
